src/foam/nanos/crunch/UpdateJunctionStatusOnDeprecatedCapabilityPut.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * @license
+ * Copyright 2019 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+foam.CLASS({
+  package: 'foam.nanos.crunch',
+  name: 'UpdateJunctionStatusOnDeprecatedCapabilityPut',
+
+  documentation: 'Set the status of usercapabilityjunctions to DEPRECATED when a capability is deprecated by another',
```

Comment:
Is this needed? In general, we seem to be proactively updating objects when other objects change, but in the implies check we can be just lazily doing the validation. This prevents objects from getting out of sync/corrupted, and is easier to code. We don't really care that is less efficient, because this will all be cached and is a one-time expense anyway.